cabal-install/src/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -93,11 +104,127 @@ import Distribution.Simple.Command
+            res <- parseProjectSkeleton cacheDir httpTransport verbosity importLoc =<< fetchImportConfig importLoc
+            rest <- go [] xs
+            pure . fmap mconcat . sequence $ [fs, res, rest]
+         (ParseUtils.Section l "if" p xs') -> do
+                           subpcs <- go [] xs'
+                           let fs = fmap singletonProjectConfigSkeleton $ fieldsToConfig (reverse acc)
+                           (elseClauses, rest) <- parseElseClauses xs
+                           let condNode =  (\c pcs e -> CondNode mempty mempty [CondBranch c pcs e]) <$>
+                                  -- we rewrap as as a section so the readFields lexer of the conditional parser doesn't get confused
+                                  adaptParseError l (parseConditionConfVarFromClause . BS.pack $ "if(" <> p <> ")") <*>
+                                  subpcs <*>
+                                  elseClauses
+                           pure . fmap mconcat . sequence $ [fs, condNode, rest]
+         _ -> go (x:acc) xs
+    go acc [] = pure . fmap singletonProjectConfigSkeleton . fieldsToConfig $ reverse acc
+
+    parseElseClauses :: [ParseUtils.Field] -> IO (ParseResult (Maybe ProjectConfigSkeleton), ParseResult ProjectConfigSkeleton)
+    parseElseClauses x = case x of
+         (ParseUtils.Section _l "else" _p xs':xs) -> do
+               subpcs <- go [] xs'
+               rest <- go [] xs
+               pure (Just <$> subpcs, rest)
+         (ParseUtils.Section l "elif" p xs':xs) -> do
+               subpcs <- go [] xs'
+               (elseClauses, rest) <- parseElseClauses xs
+               let condNode = (\c pcs e -> CondNode mempty mempty [CondBranch c pcs e]) <$>
+                                  adaptParseError l (parseConditionConfVarFromClause . BS.pack $ "else("<> p <> ")") <*>
+                                  subpcs <*>
+                                  elseClauses
+               pure (Just <$> condNode, rest)
+         _ -> (\r -> (pure Nothing,r)) <$> go [] x
+
+    fieldsToConfig xs = fmap (addProvenance . convertLegacyProjectConfig) $ parseLegacyProjectConfigFields source xs
+    addProvenance x = x {projectConfigProvenance = Set.singleton (Explicit source)}
+
+    adaptParseError _ (Right x) = pure x
+    adaptParseError l (Left e) = parseFail $ ParseUtils.FromString (show e) (Just l)
+
+    liftPR :: (a -> IO (ParseResult b)) -> ParseResult a -> IO (ParseResult b)
+    liftPR f (ParseOk ws x) = addWarnings <$> f x
+       where addWarnings (ParseOk ws' x') = ParseOk (ws' ++ ws) x'
+             addWarnings x' = x'
+    liftPR _ (ParseFailed e) = pure $ ParseFailed e
+
+    fetchImportConfig :: ProjectConfigImport -> IO BS.ByteString
+    fetchImportConfig pci = case parseURI pci of
+         Just uri -> do
+            let fp = cacheDir </> map (\x -> if isPathSeparator x then 'X' else x) (show uri) -- TODO can we do better?
```

Comment:
I think on windows, we also want to get rid of `:`, because that's not a valid filename character. So we could run [`makeValid`](https://hackage.haskell.org/package/filepath-1.4.2.2/docs/System-FilePath.html#v:makeValid) first and then git rid of the separators. Maybe `_` is better than `X` (makeValid also uses that).

So on windows:

```
> fmap (\x -> if isPathSeparator x then '_' else x) $ W.makeValid "https://some.remote.source/subdir/cabal.config"
"https___some.remote.source_subdir_cabal.config"```

Reply:
Nice suggestion -- and good point about makeValid. I'll do that.